CMakeLists.txt

```diff
@@ -195,11 +195,6 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Cray")  # cray-specific settings
 
 endif()
 
-# Get compiler flags (CFLAGS, CXXFLAGS, FFLAGS) independent of CMAKE_BUILD_TYPE
-# and echo flags that will be passed to all TPL builds
-include(get_compiler_flags)
-get_compiler_flags()
-
```

Comment:
I moved those lines to after detecting ROOT. The reason is because during ROOT detection we have disable the use of libc++ and switch to libstdc++ even if the user has configured libc++. (E.g., libc++ is the default if the compiler if clang.) This way when `get_compiler_flags()` is called, its output on the compiler flags will contain the correct flags (even after the ROOT detection has switched to libstdc++).